src/api/local-api.tsx

```diff
@@ -21,10 +21,12 @@ import { VanillaTable } from "./ui/views/vanilla-table";
 import { Callout } from "./ui/views/callout";
 import { DataArray } from "./data-array";
 import { Coerce } from "./coerce";
+import { DatacoreJSRenderer, asyncEvalInContext } from "ui/javascript";
+import { DatacoreScript, ScriptCache } from "./script-cache";
 
 /** Local API provided to specific codeblocks when they are executing. */
 export class DatacoreLocalApi {
-    public constructor(public api: DatacoreApi, public path: string) {}
+    public constructor(public api: DatacoreApi, public path: string, public loadedScripts: DatacoreScript[] = []) {}
```

Comment:
I would implement this by having `ScriptCache` contain it's own state and creating a script cache here, instead of passing the state around.